sdk/storage_blobs/src/clients/container_client.rs

```diff
@@ -15,30 +19,80 @@ use azure_storage::{
 };
 use time::OffsetDateTime;
 
-pub trait AsContainerClient {
-    fn container_client(&self, container_name: impl Into<String>) -> ContainerClient;
+#[derive(Clone, Debug)]
```

Comment:
This obviously still doesn't handle the cloud location aspect of things, but I do think it might simplify the process of adding that in.